### ISSUE TITLE
Using realm in my app, I can publish my app on the iOS AppStore, but when I upload the app to the Mac AppStore it reports an error.

### ISSUE THREAD

**Issue by whaleHan** [realm/realm-dart#1153]
### What happened?

After I upload the App to TestFight, it generates an error :

ITMS-90286: Invalid Code Signing Entitlements - Your application bundle's signature contains code signing entitlements that are not supported on macOS. Specifically, value '' for key 'com.apple.application-identifier' in 'NowDO.app/Contents/Frameworks/realm.framework/Versions/A/Resources/librealm_dart.dylib' is not supported. This value should be a string starting with your TEAMID, followed by a dot '.', followed by the bundle identifier .

### Repro steps

After I upload the App to TestFight, it generates an error 

### Version

1.0.0

### What Atlas Services are you using?

Atlas Device Sync

### What type of application is this?

Flutter Application

### Client OS and version

MacOS

**Comment by desistefanova** [comment]
Hi @whaleHan,
This must be an issue related with XCode project settings.
It is a common issue and you can find many references searching in Google.
If you have set your profile and certificates correctly and you don't have any red issues in `Signing & Capabilities` options, the deployment should be successful. We don't have any issue with the deployment. Do you have some expired certificates?
Did you try to clean and build again?
Go to XCode menu -> Product>Clean Build Folder,
then Priduct->Build and Product->Archive again.
When you build make sure you have selected "Any iOS devices".
If this doesn't help, please share your XCode Runner Settings if it is possible?

**Comment by blagoev** [comment]
Hi, 
we have investigated and found an issue related to your problem. We will be fixing it.

Cheers

**Comment by blagoev** [comment]
Hi we have merged the fix for this and it will be part of our next release very soon. 
Thanks for reporting.

**Comment by blagoev** [comment]
@whaleHan We have just released the fix for this in realm@1.0.1

**Comment by dogiaplinh** [comment]
@blagoev I still get this error on version 1.0.2

**Comment by whaleHan** [comment]
@blagoev
This problem is still present on version 1.0.2

**Comment by daikini** [comment]
@blagoev
I'm also still getting this error on version 1.0.2

**Comment by daikini** [comment]
Having an empty value in`com.apple.application-identifier` in `liberal_dart.dylib` seems to be what's causing the issue. I'm not entirely sure that this key should be here at all.

```
According to the provisioning profile, the bundle contains a key value that is not allowed: '' for the key 'com.apple.application-identifier' in '<redacted>.app/Contents/Frameworks/librealm_dart.dylib'
```
<img width="748" alt="Screenshot 2023-03-04 at 1 57 51 PM" src="https://user-images.githubusercontent.com/232/222929115-9b707b35-e7f1-47c8-91b9-430007934ca1.png">

**Comment by blagoev** [comment]
Thanks for the report. We will investigate more.

**Comment by blagoev** [comment]
A quick question: Did anyone try doing `flutter clean` after upgrading the realm dependency?

**Comment by daikini** [comment]
I usually run `flutter clean` before I run `flutter build macos --release`.

To add more context I started out with realm 0.8.0+rc and this issue was not present in that version. Attached is a screenshot of the signing dialog showing that the entitlements are not present for `librealm_dart.dylib`.

I upgraded to 0.9.0+rc to see if the entitlements were present and they were not.
I then updated to 0.10.0+rc to check the same thing and the entitlements first started showing up in this version. 

It seems like something changed between 0.9.0+rc and 0.10.0+rc with regards to the signing.

<img width="747" alt="Screenshot 2023-03-06 at 8 30 13 AM" src="https://user-images.githubusercontent.com/232/223158528-ec0c1f63-f848-4f85-9386-2d3f72c0e9b4.png">
